src/coreclr/jit/importer.cpp

```diff
@@ -20206,6 +20206,17 @@ void Compiler::impInlineRecordArgInfo(InlineInfo*   pInlineInfo,
             return;
         }
     }
+    else
+    {
+        if (curArgVal->IsCall() &&
+            (curArgVal->AsCall()->gtCallMethHnd == eeFindHelper(CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPE)) &&
+            (gtGetHelperArgClassHandle(curArgVal->AsCall()->gtArgs.GetArgByIndex(0)->GetEarlyNode()) !=
+             NO_CLASS_HANDLE))
```

Comment:
^ will be replaced with `gtIsTypeof` from https://github.com/dotnet/runtime/pull/71685/files